packaging_automation/test_citus_package.py

```diff
@@ -0,0 +1,108 @@
+    el_8 = {"name": "el/8", "docker_image_name": "el-8"}
+    centos_8 = {"name": "centos/8", "docker_image_name": "centos-8"}
+    centos_7 = {"name": "centos/7", "docker_image_name": "centos-7"}
+    ol_7 = {"name": "ol/7", "docker_image_name": "ol-7"}
+    ol_8 = {"name": "ol/8", "docker_image_name": "ol-8"}
+    debian_buster = {"name": "debian/buster", "docker_image_name": "debian-buster", }
+    debian_bullseye = {"name": "debian/bullseye", "docker_image_name": "debian-bullseye"}
+    debian_stretch = {"name": "debian/stretch", "docker_image_name": "debian-stretch"}
+    ubuntu_bionic = {"name": "ubuntu/bionic", "docker_image_name": "ubuntu-bionic"}
+    ubuntu_focal = {"name": "ubuntu/focal", "docker_image_name": "ubuntu-focal"}
+    undefined = {"name": "undefined", "docker_image_name": "undefined"}
+
+
+def parse_os_release(os_release: str) -> TestPlatform:
+    result = TestPlatform.undefined
+    for tp in TestPlatform:
+        if tp.value["name"] == os_release:
+            result = tp
+    return result
+
+
+def get_postgres_versions_from_matrix_file(project_version: str):
+    r = requests.get(POSTGRES_MATRIX_WEB_ADDRESS, allow_redirects=True)
+
+    with open(POSTGRES_MATRIX_FILE, 'wb') as writer:
+        writer.write(r.content)
+    pg_versions = get_supported_postgres_release_versions(POSTGRES_MATRIX_FILE, project_version)
+    # Since string expressions with double quotes cause problems in pipeline, string is converted into int
+    # pg_versions = [int(pg_ver) for pg_ver in pg_versions]
+
+    return pg_versions
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--prj_ver', required=True)
+    parser.add_argument('--pg_major_version')
+    parser.add_argument("--os_release", choices=[t.value["name"] for t in TestPlatform])
+
+    args = parser.parse_args()
+    test_platform = parse_os_release(args.os_release)
+    minor_prj_ver = get_minor_version(args.prj_ver)
+
+    platform = args.os_release
+
+    postgres_versions = get_postgres_versions_from_matrix_file(args.prj_ver)
+
+    print("Pg Versions: ")
+    print(postgres_versions)
+
+    os.chdir("test-images")
+    return_codes = {}
+
+    if args.pg_major_version:
+        postgres_versions = [p for p in postgres_versions if p == args.pg_major_version]
+
+    print(postgres_versions)
+    if len(postgres_versions) == 0:
+        raise ValueError("At least one supported postgres version is required")
+
+    for postgres_version in postgres_versions:
+        docker_image_name = f"test:{test_platform.value['docker_image_name']}-{postgres_version}"
+        build_command = (f"docker build -t {docker_image_name} "
+                         f"-f {test_platform.value['docker_image_name']}/Dockerfile "
+                         f"--build-arg CITUS_VERSION={args.prj_ver} --build-arg PG_MAJOR={postgres_version}   "
+                         f"--build-arg CITUS_MAJOR_VERSION={minor_prj_ver} .")
```

Comment:
You can actually print logs by doing something like that. Let me share a snippet from one of my pet projects:
```python
# Suppose "container" is of type "class Container(Model):" from
# "docker/models/containers.py".

# Log docker run output synchronously. That means, this loop never
# ends as long as container is alive or user sends a ctrl+c interrupt.
# Note that "container.logs" is a "generator" object here.
for line in container.logs(stream=True):
  print(line.decode('utf-8'))
```

Can you open an issue for that @gurkanindibay, so that we can improve this piece by following either of those approaches, or maybe something different ?

Reply:
You can actually print logs by doing something like that. Let me share a snippet from one of my pet projects:
```python
# Suppose "container" is of type "class Container(Model):" from
# "docker/models/containers.py".

# Log docker run output synchronously. That means, this loop never
# ends as long as container is alive or user sends a ctrl+c interrupt.
# Note that "container.logs" is a "generator" object here.
for line in container.logs(stream=True):
  print(line.decode('utf-8'))
```

Can you open an issue for that @gurkanindibay, so that we can improve this piece by following either of those approaches, or maybe something different ?